comp/rdnsquerier/impl/rdnsquerier.go

```diff
@@ -181,6 +183,63 @@ func (q *rdnsQuerierImpl) GetHostname(ipAddr []byte, updateHostnameSync func(str
 	return nil
 }
 
+// GetHostnameSync attempts to resolve the hostname for the given IP address synchronously.
+// If the IP address is invalid then an error is returned.
+// If the IP address is not in the private address space then it is ignored - no lookup is performed and nil error is returned.
+// If the IP address is in the private address space then the IP address will be resolved to a hostname.
+// The function accepts a timeout duration, which defaults to 2 seconds if not provided.
+func (q *rdnsQuerierImpl) GetHostnameSync(ctx context.Context, ipAddr string) (string, error) {
+	q.logger.Infof("ENTERED THE GetHostnameSync FUNCTION with IP address: %s", ipAddr)
+
+	q.internalTelemetry.total.Inc()
+
+	netipAddr := net.ParseIP(ipAddr).To4()
+	if netipAddr == nil {
+		q.internalTelemetry.invalidIPAddress.Inc()
+		return "", fmt.Errorf("invalid IP address %v", ipAddr)
+	}
+
+	if !netipAddr.IsPrivate() {
```

Comment:
Do we need this check here? I think we already do it in `GetHostname` Do we want to return an error for it? For Network Path my thought was just to throw every IP at the function and expect that the hostname would be blank if it's not a private IP

Reply:
1/ I think the rdnsquerier library shouldn't be responsible for filtering specific IPs e.g. Private IPs, and leave that responsibility to the caller.

My rational is that it makes the library more rigid, what if a client wants to query rDNS for public IPs?

Alternatively, it could be an option passed to `GetHostnameSync()`. Maybe using the options pattern: https://dev.to/kittipat1413/understanding-the-options-pattern-in-go-390c

2/ I thought we were already filtering out private IPs for NetFlow?